src/Model/Qualification.php

```diff
@@ -32,44 +32,85 @@ class Qualification {
 	private int $month;
 	private int $day;
 
+	private array $undefinedParts = [];
+	private array $uncertainParts = [];
+	private array $approximateParts = [];
+	private array $uncertainAndApproximateParts = [];
+
 	public function __construct( int $year = 0, int $month = 0, int $day = 0 ) {
 		// TODO: Does it make sense these constructor params are optional?
 		$this->year = $year;
 		$this->month = $month;
 		$this->day = $day;
+
+		$this->setUncertainty();
+	}
+
+	private function setUncertainty(): void {
+		foreach ( [ 'year' => $this->year, 'month' => $this->month, 'day' => $this->day ]
+			as $part => $value ) {
+
+			switch( $value ) {
+				case self::UNDEFINED:
+					$this->undefinedParts[] = $part;
+					break;
+				case self::UNCERTAIN :
+					$this->uncertainParts[] = $part;
+					break;
+				case self::APPROXIMATE :
+					$this->approximateParts[] = $part;
+					break;
+				case self::UNCERTAIN_AND_APPROXIMATE :
+					$this->uncertainAndApproximateParts[] = $part;
+					break;
+			}
+		}
+	}
+
+	public function getWholeDateUncertainty(): int {
+		if ( count( $this->uncertainParts ) + count( $this->undefinedParts ) === 3 ) {
+			return self::UNCERTAIN;
+		}
+
+		if ( count( $this->approximateParts ) + count( $this->undefinedParts ) === 3 ) {
+			return self::APPROXIMATE;
+		}
+
+		if ( count( $this->uncertainAndApproximateParts ) + count( $this->undefinedParts ) === 3 ) {
+			return self::UNCERTAIN_AND_APPROXIMATE;
+		}
+
+		return -1;
+	}
+
+	public function getUndefinedParts(): array {
```

Comment:
`@return string[]`

Also good use use constants for the day month and year strings.

Reply:
I'm not sure I understand to what is referred the following

> @return string[]

regarding constants for day month and year strings, perhaps we can leave as strings considered that in the same class they are used as strings in other methods ? (unless you don't want to distinguish them)